### PR TITLE
Mojo origin cleanup

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/31984

- No longer use origin as an Id for some requests. Using random string instead.
- AddPermission, ResetPermission, etc now use current active origin.
- For OriginInfo spec and eTldPlusOne strings are enough.
- For android removed origin tracking. It just needs OriginInfo(spec + eTldPlusOne) and whether current activity is panel or wallet to access either per-active-origin or default network APIs
- There some network APIs left which need origin args for ios needs. Will be handled in next PR.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by darkdh — APPROVED** [review]
Nice cleanup

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java [hunk 1]
```diff
@@ -265,41 +266,39 @@ public boolean isShowing() {
...
-        if (CoinType.SOL == selectedAccount.accountId.coin) {
+    private void updateSelectedAccountConnectionState() {
+        if (CoinType.SOL == mSelectedAccount.accountId.coin) {
             if (mAccountsWithPermissions.size() == 0) {
                 mBtnConnectedStatus.setText("");
                 mBtnConnectedStatus.setVisibility(View.GONE);
                 mCvSolConnectionStatus.setVisibility(View.GONE);
             } else {
-                isSolanaConnected(selectedAccount);
+                isSolanaConnected(mSelectedAccount);
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:295]: nit: `isSolanaConnected` should be renamed to `updateSolanaConnected`, might have been missed during initial review.

File: android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java [hunk 2]
```diff
@@ -129,19 +136,8 @@ public NetworkModel(BraveWalletService braveWalletService, JsonRpcService jsonRp
...
-                if (TextUtils.isEmpty(chainId)) {
-                    mJsonRpcService.setNetwork(
-                            id, mSharedData.getCoinType(), mOrigin, hasSetNetwork -> {});
-                } else {
-                    id = chainId;
-                }
-                _mChainId.postValue(id);
-            });
-        });
+        _mChainId.addSource(mMode, mode -> { updateChainId(); });
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java:139]: We don't need mode as a live data, just keeping a reference should work , `updateChainId()` should be called when `updateMode` is called.

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java (hunk 1 above)
Comment (supermassive) [reply android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java]: https://github.com/brave/brave-core/pull/19329/commits/2b62a3db430107196b574a0d05990d585d0ee963

File: android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java (hunk 2 above)
Comment (supermassive) [reply android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java]: https://github.com/brave/brave-core/pull/19329/commits/2b62a3db430107196b574a0d05990d585d0ee963

**Review by Pavneet-Sing — APPROVED** [review]

**Review by onyb — APPROVED** [review]
frontend++

**Review by StephenHeaps — APPROVED** [review]
iOS++. iOS changes available on [wallet/mojo-origin-cleanup](https://github.com/brave/brave-ios/compare/brave-core-1.58.x...wallet/mojo-origin-cleanup) branch.

> There some network APIs left which need origin args for ios needs. Will be handled in next PR.

Friendly reminder we lack `GetActiveOrigin` and some other delegate methods on iOS at the moment: https://github.com/brave/brave-browser/issues/31395

**Pull request merged**